cmd/dtn-tool/bpsec.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	err = b.AddExtensionBlock(eb)
+	if err != nil {
+		printFatal(err, "Add Extension Block failed")
+
+	}
+
+	bibBlockAdded, _ := b.ExtensionBlock(bib.BlockTypeCode())
+
+	err = bibBlockAdded.Value.(*bpv7.BIBIOPHMACSHA2).SignTargets(b, bibBlockAdded.BlockNumber, []byte(psk))
+	if err != nil {
+		return
+	}
+
+	logger := log.WithFields(log.Fields{
+		"bundle": b.ID(),
+		"file":   output,
+	})
+
+	if f, err := os.Create(output); err != nil {
+		logger.WithError(err).Error("Creating file errored")
+		return
+	} else if err := b.MarshalCbor(f); err != nil {
+		logger.WithError(err).Error("Marshalling Bundle errored")
+	} else if err := f.Close(); err != nil {
+		logger.WithError(err).Error("Closing file errored")
+	}
+
+}
+
+// verifyBundle for the "verify" CLI option.
+func verifyBundle(args []string) {
+	if len(args) != 2 {
+		printUsage()
+	}
+
+	var (
+		input    = args[0]
+		psk      = args[1]
+		err      error
+		f        io.ReadCloser
+		b        bpv7.Bundle
+		bibBlock *bpv7.CanonicalBlock
+	)
+
+	if input == "-" {
+		f = os.Stdin
+	} else if f, err = os.Open(input); err != nil {
+		printFatal(err, "Opening file for reading errored")
+	}
+
+	if err = b.UnmarshalCbor(f); err != nil {
+		printFatal(err, "Unmarshaling Bundle errored")
+	}
+	if err = f.Close(); err != nil {
+		printFatal(err, "Closing file errored")
+	}
+
+	bibBlock, err = b.ExtensionBlock(bpv7.ExtBlockTypeBlockIntegrityBlock)
+	if err != nil {
+		printFatal(err, "Could not get BIB Extension Block")
+	}
+
+	err = bibBlock.Value.(*bpv7.BIBIOPHMACSHA2).VerifyTargets(b, bibBlock.BlockNumber, []byte(psk))
+	if err != nil {
+		printFatal(err, "Verfication Error")
+	}
+
+	log.Info("Verify OK")
+
+}
+
+// encryptBundle for the "sign" CLI option.
+func encryptBundle(args []string) {
+	if len(args) != 3 {
+		printUsage()
+	}
+
+	var (
+		input  = args[0]
+		psk    = args[1]
+		output = args[2]
+		err    error
+		f      io.ReadCloser
+		b      bpv7.Bundle
+	)
+
+	if input == "-" {
+		f = os.Stdin
+	} else if f, err = os.Open(input); err != nil {
+		printFatal(err, "Opening file for reading errored")
+	}
+
+	if err = b.UnmarshalCbor(f); err != nil {
+		printFatal(err, "Unmarshaling Bundle errored")
+	}
+	if err = f.Close(); err != nil {
+		printFatal(err, "Closing file errored")
+	}
+
+	payloadSecurityTarget, _ := b.ExtensionBlock(bpv7.ExtBlockTypePayloadBlock)
+
+	aesVariant := bpv7.A256GCM
+
+	bcb := bpv7.NewBCBIOPAESGCM(&aesVariant, nil, nil, payloadSecurityTarget.BlockNumber, b.PrimaryBlock.SourceNode)
+
+	eb := bpv7.CanonicalBlock{
+		BlockNumber:       0,
+		BlockControlFlags: 0,
+		CRCType:           bpv7.CRCNo,
+		CRC:               nil,
+		Value:             bcb,
+	}
+
+	err = b.AddExtensionBlock(eb)
+	if err != nil {
+		printFatal(err, "Add Extension Block failed")
+
+	}
+
+	bcbBlockAdded, _ := b.ExtensionBlock(bcb.BlockTypeCode())
+
+	err = bcbBlockAdded.Value.(*bpv7.BCBIOPAESGCM).EncryptTarget(b, bcbBlockAdded.BlockNumber, []byte(psk))
+	if err != nil {
```

Comment:
Log the error?

Reply:
Btw, this fails exactly at this point for me:
```
$ ./dtn-tool encrypt signout uwu encryptout
./dtn-tool errored: EncryptTarget failed
  keylength 3 is not supported
```
When this works. Thus, the error should be printed:
```
$ ./dtn-tool encrypt signout uwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwu encryptout
```

Reply:
k done